docs: add drag and drop guide

andrewseguin opened the pull request:
Add new angular.dev documentation for the CDK drag-and-drop. This is part of the process of migrating the docs from material.angular.io

This was originally opened in https://github.com/angular/angular/pull/58232

amysorto committed:
docs: add drag and drop guide

amysorto committed:
docs: add code examples for drag and drop guide

# Conflicts:
#	adev/src/content/guide/BUILD.bazel

mmalerba commented:
caretaker note: merge together with the PRs that add the API reference for drag & drop:
https://github.com/angular/angular/pull/61346
https://github.com/angular/angular/pull/61347
https://github.com/angular/angular/pull/61348

mmalerba committed:
ci: fix build

Deletes the BUILD.bazel in examples/drag-drop so the files can be
collected in examples/BUILD.bazel instead

andrewseguin committed:
docs: add preview to an example to test

mmalerba committed:
docs: fix embedded preview

update the embeddable examples glob to captrue the drag & drop examples,
and fix the current build issues in the drag & drop examples

andrewseguin committed:
docs: make drag and drop examples preview

mmalerba reviewed (approved):
I flipped through and the embedded examples all look good now

thePunderWoman commented:
This PR was merged into the repository by commit 25317dd958f0fe2a516f66315235e1843ef8cb37.

The changes were merged into the following branches: main, 19.2.x, 20.0.x
